File: java/CommandInjection.java
import java.util.Scanner;

public class CommandInjection {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter command: ");
        String cmd = sc.nextLine();

        // CWE-78: user input passed to Runtime.exec via shell without validation.
        Runtime.getRuntime().exec(new String[]{"/bin/sh", "-c", cmd});
    }
}

File: csharp/CommandInjection.cs
using System;
using System.Diagnostics;
using System.Collections.Generic;

namespace SastSamples
{
    public static class CommandInjection
    {
        public static void Main()
        {
            Console.Write("Enter command (list/date/help): ");
            var cmd = Console.ReadLine();

            // Map user input to safe, predefined commands
            var allowedCommands = new Dictionary<string, Action>
            {
                ["list"] = () => Process.Start(new ProcessStartInfo("/bin/ls", "-l /tmp") { UseShellExecute = false }),
                ["date"] = () => Process.Start(new ProcessStartInfo("/bin/date") { UseShellExecute = false }),
                ["help"] = () => Console.WriteLine("Allowed commands: list, date, help")
            };

            if (allowedCommands.TryGetValue(cmd?.ToLower() ?? "", out var action))
            {
                action();
            }
            else
            {
                Console.WriteLine("Invalid command. Type 'help' for allowed commands.");
            }
        }
    }
}
